metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -967,6 +968,10 @@ public void handleLeaderChange(LeaderAndEpoch newLeader) {
                         metadataVersion = featureControl.metadataVersion();
                     }
 
+                    curClaimEpoch = newEpoch;
```

Comment:
Hmm, why is it necessary to reorder these fields?

Reply:
This is primarily to fix the race in the test code. If we update the volatile sooner, there is a possibility the test code will see the controller as active, enqueue the brokerRegister event, and process the event before the bootstrap metadata gets enqueued.